opennms-config-jaxb/src/main/java/org/opennms/netmgt/config/snmp/Definition.java

```diff
@@ -184,10 +197,23 @@ public boolean equals(final Object obj) {
         } else if (!m_specifics.equals(other.m_specifics)) {
             return false;
         }
+        if (m_location == null) {
+            if (other.m_location != null) {
+                return false;
+            }
+        } else if (!m_location.equals(other.m_location)) {
+            return false;
+        }
+
         return true;
     }
 
     public void visit(final SnmpConfigVisitor visitor) {
+
+        if ((visitor.getLocation() != null) && (!visitor.getLocation().equals(m_location))) {
```

Comment:
This logic is in the wrong place, it should be up to the visitor implementation to decide if it wants to visit a definition or not.